Source/ARTLocalDeviceStorage.m

```diff
@@ -7,16 +7,65 @@
 //
 
 #import "ARTLocalDeviceStorage.h"
+#import "ARTLog.h"
+#import "ARTLocalDevice+Private.h"
+#import <SAMKeychain/SAMKeychain.h>
 
-@implementation ARTLocalDeviceStorage
+@implementation ARTLocalDeviceStorage {
+    __weak ARTLog *_logger;
+}
+
+- (instancetype)initWithLogger:(ARTLog *)logger {
+    if (self = [super init]) {
+        _logger = logger;
+    }
+    return self;
+}
+
++ (instancetype)newWithLogger:(ARTLog *)logger {
```

Comment:
This seems inessential. You can use the usual alloc/init when you call it here: https://github.com/ably/ably-ios/blob/push-device-auth/Source/ARTRest.m#L92